File: leetcode_solutions/0392-is-subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        p_s = 0
        p_t = 0
        substring = ""

        while (p_t < len(t) and p_s < len(s)):
            if s[p_s] == t[p_t]:
                substring += t[p_t]
                p_s += 1
            p_t += 1
        
        return substring == s

File: leetcode_solutions/0392-is-subsequence/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int pS = 0;
        int pT = 0;
        while (pS < s.length() && pT < t.length()){
            if (s.at(pS) == t.at(pT++)){
                pS++;
            }
        }
        return pS == s.length() ? true : false;
    }
};
